### PR TITLE
Refactor DeepSeek logic into atomic operations

### PR DESCRIPTION
**Pull request by fzyzcjy**
## Motivation

For unified TBO

Please subtract diff from previous PRs

Tests

```
SGL_ENABLE_JIT_DEEPGEMM=0 python3 -m sglang.launch_server --model-path deepseek-ai/DeepSeek-V2-Lite --trust-remote-code --tp 4 --dp 2 --host 0.0.0.0 --port 12321 --enable-dp-attention --max-running-requests 2048 --disable-cuda-graph --enable-deepep-moe --deepep-mode normal
CUDA_VISIBLE_DEVICES=4,5,6,7 SGL_ENABLE_JIT_DEEPGEMM=0 python3 -m sglang.launch_server --model-path deepseek-ai/DeepSeek-V2-Lite --trust-remote-code --tp 4 --dp 4 --host 0.0.0.0 --port 22222 --enable-dp-attention --max-running-requests 2048 --disable-cuda-graph --enable-deepep-moe --deepep-mode normal
SGL_ENABLE_JIT_DEEPGEMM=0 python3 -m sglang.launch_server --model-path deepseek-ai/DeepSeek-V2-Lite --trust-remote-code --tp 2 --dp 2 --host 0.0.0.0 --port 12321 --max-running-requests 2048 --disable-cuda-graph --enable-deepep-moe --deepep-mode normal
SGL_ENABLE_JIT_DEEPGEMM=0 python3 -m sglang.launch_server --model-path deepseek-ai/DeepSeek-V2-Lite --trust-remote-code --tp 4 --host 0.0.0.0 --port 12321 --max-running-requests 2048
CUDA_VISIBLE_DEVICES=4,5,6,7 SGL_ENABLE_JIT_DEEPGEMM=0 python3 -m sglang.launch_server --model-path deepseek-ai/DeepSeek-V2-Lite --trust-remote-code --tp 4 --dp 4 --enable-dp-attention --host 0.0.0.0 --port 22222 --max-running-requests 2048 --disable-cuda-graph

cd /host_home/primary_synced/sglang/benchmark/mmlu
bash download_data.sh
while true; do python3 bench_sglang.py --nsub 99999 --port 12321; done
```

All are 57.7-58.0 mmlu which aligns with baseline

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/operations_strategy.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import torch
+
+
+def compute_layer_operations(
```
Comment (yizhang2077) [inline python/sglang/srt/operations_strategy.py:4]: Can we set strategy as model specific? each model can register own stategy and this can be default strategy.

Comment (fzyzcjy) [reply python/sglang/srt/operations_strategy.py:4]: since I do not know whether qwen will have a different strategy or same strategy, I choose to make it minimalist here. But definitely we should make it more complex when we find we need several strategies.

Indeed I guess we will need e.g. 2 strategies for 3 models, i.e. strategy and model may not be 1:1. And for different hardware it may be even possible to need diff strategy.

Comment (yizhang2077) [reply python/sglang/srt/operations_strategy.py:4]: For example, qwen3 has no shared_expert, so `op_shared_expert` will not be used in qwen3

Comment (yizhang2077) [reply python/sglang/srt/operations_strategy.py:4]: yes, so I think it is better to can be configurable, but this pr is only for refactor and I think current way it is ok.

Comment (fzyzcjy) [reply python/sglang/srt/operations_strategy.py:4]: totally agree

**Comment by yizhang2077** [comment]
I think we need merge #6321 first?

**Comment by fzyzcjy** [comment]
@yizhang2077 Yes, we need the 4 PRs as prerequisite: 6316, 6321, 6324, 6325, before merging this one

**Comment by fzyzcjy** [comment]
(I will resolve conflicts after prereq are merged since that will introduce more conflicts)

**Review by sleepcoo — APPROVED** [review]
